src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs

```diff
@@ -105,18 +105,52 @@ public override void OnCustom(string name, Activity activity, object payload)
                             case "Microsoft.EntityFrameworkCore.Sqlite":
+                            case "Devart.Data.SQLite.EFCore":
                                 activity.AddTag(AttributeDbSystem, "sqlite");
                                 break;
                             case "MySql.Data.EntityFrameworkCore":
                             case "Pomelo.EntityFrameworkCore.MySql":
+                            case "Devart.Data.MySql.EFCore":
                                 activity.AddTag(AttributeDbSystem, "mysql");
                                 break;
                             case "Npgsql.EntityFrameworkCore.PostgreSQL":
+                            case "Devart.Data.PostgreSql.EFCore":
                                 activity.AddTag(AttributeDbSystem, "postgresql");
                                 break;
                             case "Oracle.EntityFrameworkCore":
+                            case "Devart.Data.Oracle.EFCore":
                                 activity.AddTag(AttributeDbSystem, "oracle");
                                 break;
+                            case "Microsoft.EntityFrameworkCore.InMemory":
+                                activity.AddTag(AttributeDbSystem, "efcore_in_memory");
+                                break;
+                            case "FirebirdSql.EntityFrameworkCore.Firebird":
+                                activity.AddTag(AttributeDbSystem, "firebird");
+                                break;
+                            case "FileContextCore":
+                                activity.AddTag(AttributeDbSystem, "filecontextcore");
+                                break;
+                            case "EntityFrameworkCore.SqlServerCompact35":
+                            case "EntityFrameworkCore.SqlServerCompact40":
+                                activity.AddTag(AttributeDbSystem, "mssqlcompact");
+                                break;
+                            case "EntityFrameworkCore.OpenEdge":
+                                activity.AddTag(AttributeDbSystem, "openedge");
+                                break;
+                            case "EntityFrameworkCore.Jet":
+                                activity.AddTag(AttributeDbSystem, "jet");
+                                break;
+                            case "Google.Cloud.EntityFrameworkCore.Spanner":
+                                activity.AddTag(AttributeDbSystem, "spanner");
+                                break;
+                            case "Teradata.EntityFrameworkCore":
+                                activity.AddTag(AttributeDbSystem, "teradata");
+                                break;
+                            case "IBM.EntityFrameworkCore":
+                            case "IBM.EntityFrameworkCore-lnx":
+                            case "IBM.EntityFrameworkCore-osx":
+                                activity.AddTag(AttributeDbSystem, "ibm");
```

Comment:
This one is an open question imho. Id'ing it as an ibm system seems better than defaulting outright to other_sql.